src/App.js

```diff
@@ -1,13 +1,105 @@
 import React, { Component } from 'react'
+import PropTypes from 'prop-types'
 import { Instructions } from './components'
 import './styles/app.css'
 
+class Counter extends Component {
+  static propTypes = {
+    initialCount: PropTypes.number.isRequired
+  }
+  constructor(props) {
+    super(props)
+    this.state = { count: this.props.initialCount }
+  }
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      count: nextProps.initialCount
+    })
+  }
+  incClick = () => {
+    this.setState(state => ({
+      count: state.count + 1
+    }))
+  }
+  decClick = () => {
+    this.setState(state => ({
+      count: state.count - 1
+    }))
+  }
+  render() {
+    return (
+      <div>
+        Count state: {this.state.count}
+        <button onClick={this.decClick}>Decrease</button>
+        <button onClick={this.incClick}>Increase</button>
+      </div>
+    )
+  }
+}
+
+class InitialCountForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      count: 0,
+      newCount: 0
+    }
+  }
+  handleSubmit(event) {
+    this.setState({
+      count: Number(this.state.newCount),
+      newCount: this.state.newCount
+    })
+    this.props.appCallback(Number(this.state.newCount))
+  }
+  handleForm(event) {
+    this.setState({
+      count: Number(this.state.count),
+      newCount: event.target.value
+    })
+  }
+  render() {
+    return (
+      <div>
+        <label>
+          Set count state:
+          <input
+            type="text"
+            placeholder="Set count state"
+            onChange={this.handleForm.bind(this)}
```

Comment:
Instead of using `.bind`, use ES6 autobinding.
i.e. define `handleForm` as an arrow function property similar to what you did for `decClick` in  `Counter`, and then you can use it here without need to `bind this`

Here's some reading: https://www.ian-thomas.net/autobinding-react-and-es6-classes/